Formula/z/zig.rb

```diff
@@ -22,15 +22,23 @@ class Zig < Formula
   end
 
   depends_on "cmake" => :build
-  depends_on "llvm@15" => :build
+  depends_on "llvm" => :build
   depends_on macos: :big_sur # https://github.com/ziglang/zig/issues/13313
   depends_on "zstd"
   uses_from_macos "ncurses"
   uses_from_macos "zlib"
 
-  fails_with gcc: "5" # LLVM is built with GCC
+  fails_with :gcc
 
   def install
+    # Workaround for https://github.com/Homebrew/homebrew-core/pull/141453#discussion_r1320821081.
+    # This will likely be fixed upstream by https://github.com/ziglang/zig/pull/16062.
+    if OS.linux?
+      ENV["NIX_LDFLAGS"] = ENV["HOMEBREW_RPATH_PATHS"].split(":")
```

Comment:
what if "HOMEBREW_RPATH_PATHS" is not set? Will `ENV["HOMEBREW_RPATH_PATHS"` be null (nil) and therefore `.split` on will lead to null reference exceptions?

Reply:
It's defined here, but on Linux only: https://github.com/Homebrew/brew/blob/502434cf440d6dc900d99414ef95ecde57859221/Library/Homebrew/extend/os/linux/extend/ENV/super.rb#L22

The `NilClass` exception would occur without the `if OS.linux?`, though.

Reply:
you update the pr faster than I can think – while I think about nil, you already fix the issue with non-linux :)